packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -99,9 +112,21 @@ export class OpenIdConnectProvider extends Resource implements IOpenIdConnectPro
    * @param openIdConnectProviderArn the ARN to import
    */
   public static fromOpenIdConnectProviderArn(scope: Construct, id: string, openIdConnectProviderArn: string): IOpenIdConnectProvider {
+    const parsedResourceName = Stack.of(scope).parseArn(openIdConnectProviderArn).resourceName;
+    if (!parsedResourceName) {
+      throw new Error('Cannot figure OIDC issuer url for this OIDC Provider');
+    }
+
+    // this needed because TS don't understand that prev. condition
+    // actually does mutate the type from "string | undefined" to "string"
+    // inside class definition,
+    const resourceName = parsedResourceName;
+
     class Import extends Resource implements IOpenIdConnectProvider {
       public readonly openIdConnectProviderArn = openIdConnectProviderArn;
+      public readonly openIdConnectProviderIssuer = resourceName;
```

Comment:
```suggestion
      public readonly openIdConnectProviderIssuer = parsedResourceName!;
```

Reply:
ah I so much dislike using "!" inside typescript. even here, in this obvious place.

Reply:
Absolutely agree. `!` should be avoided. You can add a null runtime check

Reply:
it is there